supplemental_notebooks/2020-07-30_pathway_analysis.Rmd

```diff
@@ -0,0 +1,702 @@
+                     # Compute Gaussian-distributed scores
+                     mx.diff = TRUE,
+                     # Use 4 cores
+                     parallel.sz = 4)
+```
+
+### Downstream analysis of GSVA scores
+
+We can test the pathway-level scores from GSVA for differential expression.
+We'll use the [`limma` package](https://bioconductor.org/packages/release/bioc/html/limma.html) to do so ([vignette](https://bioconductor.org/packages/release/bioc/vignettes/limma/inst/doc/usersguide.pdf)).
+A differential expression analysis with `limma` example is included in the GSVA documentation!
+
+_Note: `limma` is a great package for many types of differential expression analyses._ 
+_Here, we're showing you one way to approach the questions we'll lay out below._
+
+As a reminder, the PDX models in this experiment are from two different medulloblastoma subgroups (`Group` column in the metadata) and digoxin treatment (`treatment`) is being tested.
+
+We can use `limma` to identify the following:
+
+- Which pathways respond to digoxin in Group 3 PDX?
+- Which pathways respond to digoxin in Group 4 PDX?
+- Which pathways respond differently between Group 4 and Group 3 PDX?
+
+You may notice that we could only take into account one of these labels (Group) when we put together the files required for running GSEA. 
+That's a limitation of running GSEA with phenotype permutation.
+
+```{r}
+# Create a factor that combines the molecular subtype and treatment information
+subtype_treatment <- paste(metadata_df$Group, 
+                           metadata_df$treatment,
+                           sep = ".")
+subtype_treatment <- factor(subtype_treatment,
+                            levels = c("Group_3.none", 
+                                       "Group_3.digoxin",
+                                       "Group_4.none",
+                                       "Group_4.digoxin"))
+# Create a model matrix 
+design_mat <- model.matrix(~ 0 + subtype_treatment)
+colnames(design_mat) <- levels(subtype_treatment)
+
+# Fit a model with a coefficient for all four combinations of subtype and 
+# treatment -- this is fitting a linear model for each pathway
+fit <- lmFit(gsva_results, design_mat)
+```
+
+Now we're ready to extract the contrasts that answer our questions of interest.
+
+```{r}
+contrast_matrix <- makeContrasts(
+  # Response to digoxin in Group 3
+  treatment_group_3 = Group_3.digoxin-Group_3.none,
+  # Response to digoxin in Group 4
+  treatment_group_4 = Group_4.digoxin-Group_4.none,
+  # Difference in treatment response between subtypes
+  group_diff = (Group_4.digoxin-Group_4.none)-(Group_3.digoxin-Group_3.none),
+  levels = design_mat
+)
+fit2 <- contrasts.fit(fit, contrast_matrix)
+
+# Apply empirical Bayes smoothing
+fit2 <- eBayes(fit)
+```
+
+Let's take a look at the top 10 pathways returned.
+(To look just at the pathways for the Group 3 digoxin vs. no treatment, we could do `topTable(fit2, coef = 1)`, instead.)
```

Comment:
For my understanding, this is the top 10 of what? Significance vs. null of no variation across groups? 
I do think it might be useful to show one of the other contrast results as well: perhaps the difference in treatment effect by group (but that's me, I love an interaction)?